client/scala/src/test/scala/ai/verta/blobs/dataset/TestVersioning.scala

```diff
@@ -0,0 +1,80 @@
+package ai.verta.blobs.dataset
+
+import ai.verta.client._
+import ai.verta.blobs._
+import ai.verta.swagger.client.HttpException
+import ai.verta.swagger._public.modeldb.versioning.model.VersioningSetTagRequestResponse
+
+import scala.concurrent.ExecutionContext
+import scala.language.reflectiveCalls
+import scala.util.{Try, Success, Failure}
+
+import org.scalatest.FunSuite
+import org.scalatest.Assertions._
+
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3._
+import com.amazonaws.services.s3.model._
+import collection.JavaConverters._
+
+import scala.collection.mutable.HashSet
+
+class TestVersioning extends FunSuite {
+  def fixture =
+    new {
+      val testfilePath = "s3://verta-scala-test/testdir/testfile"
+      val testfileLoc = S3Location(testfilePath).get
+
+      val testfilePath2 = "s3://verta-scala-test/testdir/testsubdir/testfile2"
+      val testfileLoc2 = S3Location(testfilePath2).get
+
+      val testdirPath = "s3://verta-scala-test/testdir/"
+      val testdirLoc = S3Location(testdirPath).get
+
+      val testsubdirPath = "s3://verta-scala-test/testdir/testsubdir/"
+      val testsubdirLoc = S3Location(testsubdirPath).get
+
+      val bucketLoc = S3Location("s3://verta-scala-test").get
+
+      val workingDir = System.getProperty("user.dir")
+      val testDir = workingDir + "/src/test/scala/ai/verta/blobs/testdir"
+      val testfile = testDir + "/testfile"
+      val testSubdir = testDir + "/testsubdir"
+      val testfile2 = testSubdir + "/testfile2"
+    }
+
+  test("The downloaded s3 file should not be corrupted") {
+    val f = fixture
+    val s3Blob = S3(f.bucketLoc).get
+    val s3PathToLocalMap = s3Blob.prepareForUpload().get.mapValues(comp => comp.filePath)
```

Comment:
I'd recommend testing this at the commit level since that's what customers will use. Right now, there's some code between `commit.save` and `blob. prepareForUpload` that is not tested.

Reply:
Hmm OK. I'll remove this test and test this more carefully at Commit level when we actually implemented download. Right now I agree that this is not very helpful.

Reply:
Removed.

Reply:
@nhatsmrt please file a follow-up

Reply:
@conradoverta wait what do you mean by follow-up? Do you want me to write a test now or wait until the stuff inside Commit is implemented? I was under the impression that you just wanted me to remove this test and wait until we have more stuff to do integration tests.

Reply:
Jira ticket. We can merge this in as-is, but we should have a reminder that this needs to be properly tested once commit is in.